### ISSUE TITLE
Electron 8.x spell checker right click menu does not work

### ISSUE THREAD

**Issue by pashvin** [electron/electron#22698]
* [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [X] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 8.x
* **Operating System:**
  * Mac Catalina and Windows 10 latest
* **Last Known Working Electron version:**
  * 8.x

### Expected Behavior
On detecting wrong spelling, right click should give suggestions

### Actual Behavior
I see red underline for wrong spelling but right click does not give any menu

### To Reproduce
Go to https://github.com/electron/electron-quick-start
Get source code and modify a bit
First in main.js pass spellcheck flag in Browser window.
In Index.html, add one input box.
Run application and type wrong spelling in input box.
You will see red underline. Right click on that word.
You will see there is no right click with suggestions.
I think I am missing something simple to get context menu

**Comment by codebytere** [comment]
cc @MarshallOfSound

**Comment by MarshallOfSound** [comment]
You need to create your own context menu for any right click action.  There is a "context-menu" event on any webContents.  With that information there is a bunch of information about what was right clicked.  If a misspelled word was right click information about that word and it's dictionary suggestions will be provided in the event arguments.  Check the "context-menu" event docs for details on that.

You then use the Menu API to create and popup your context menu.

TLDR: you have to construct your own context menu, but all information is handed to you

**Comment by pashvin** [comment]
I am OK for creating own menu ( and that is what I thought) but can you point me the call back where the spell checker will give all possible suggestions so I can take those and put in my menu?

**Comment by staifan** [comment]
Hello @pashvin
I'm in the same situation as you. I used "node-spellchecker" on electronic version 5 which does not work with version 8.
Suddenly I try to update my application with the native spell checker but the operating logic is not at all the same and the current documentation is not clear on this subject because you have to navigate everywhere to find the right information ...
If it can help you or someone, I've just found out how to get the spell checker's suggestions:

> Main Process

`const window = new BrowserWindow({
                webPreferences: {
                    spellcheck: true
                }
            });`
`window.loadURL('file://mainView.html');`
`const session = window.webContents.session;`
`session.getSpellCheckerLanguages(['fr']);`
`session.setSpellCheckerLanguages(['fr']);`
`window.webContents.on('context-menu', (event, params) => {`
                `console.log('context-menu event : ',event)`
                `console.log('context-menu param : ',params)`
           ` })`
you will find the words suggested by a spell checker in params.dictionarySuggestions

Additional info: https://www.electronjs.org/docs/api/web-contents (event--context-menu)

**Comment by vladimiry** [comment]
> I used "node-spellchecker" on electronic version 5 which does not work with version 8.

It does, and with v9/v10 too, use version from this PR https://github.com/atom/node-spellchecker/issues/127

**Comment by staifan** [comment]
ok @vladimiry.
Sorry I didn't want to say anything bad about your module which has been very useful to me until now but unfortunately, I can't build it with the classic command: **electron-rebuild.cmd**.
Surely I am not doing the right thing, but I admit that I did not understand everything at the link you mentioned and especially what I need to modify for it to work.
So I prefer to try using the native electron spell checker.

**Comment by vladimiry** [comment]
> Sorry I didn't want to say anything bad about your module

That's certainly not my module. The point of the previous message was in highlighting that `node-spellchecker` still works well with Electron, including the modern versions (even `beta` and `nighty` ones).

**Comment by samuelmeuli** [comment]
[electron-context-menu](https://github.com/sindresorhus/electron-context-menu) now provides dictionary suggestions for misspelled words.

**Comment by ahallicks** [comment]
Just to add to this I've created a little Gist with an Electron Fiddle example of how to quickly and easily get spellchecker working:

https://gist.github.com/ahallicks/4bcd0bc78af624dd62bcb6416abeb714

I think the part that is important is that it only works inside elements where the text can be edited (such as a form input, or contenteditable region).

**Comment by tiffsea** [comment]
> Hello @pashvin
> I'm in the same situation as you. I used "node-spellchecker" on electronic version 5 which does not work with version 8.
> Suddenly I try to update my application with the native spell checker but the operating logic is not at all the same and the current documentation is not clear on this subject because you have to navigate everywhere to find the right information ...
> If it can help you or someone, I've just found out how to get the spell checker's suggestions:
> 
> > Main Process
> 
> ```const window = new BrowserWindow({ webPreferences: { spellcheck: true } });
> window.loadURL('file://mainView.html');
> const session = window.webContents.session;
> session.getSpellCheckerLanguages(['fr']);
> session.setSpellCheckerLanguages(['fr']);
> window.webContents.on('context-menu', (event, params) => {
> console.log('context-menu event : ',event)
> console.log('context-menu param : ',params)
> })

> you will find the words suggested by the spell checker in params.dictionarySuggestions
> 
> Additional info: https://www.electronjs.org/docs/api/web-contents (event--context-menu)

This is super helpful and works for me. I'm curious though, I'm sure it exists somewhere but what is the code for allowing user to select one of these `params.dictionarySuggestions` and repopulating app fields?
Thank you!

**Comment by MarshallOfSound** [comment]
@tiffsea Check out our spellchecker docs --> https://www.electronjs.org/docs/tutorial/spellchecker#how-do-i-put-the-results-of-the-spellchecker-in-my-context-menu

There's a complete code sample in there
